api/v1/hostdefinition_types.go

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright 2019 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// HostDefinitionSpec defines the desired state of HostDefinition
+type HostDefinitionSpec struct {
+	HostDefinition IBMBlockCSIHostDefinitionSpec `json:"hostDefinition"`
+}
+
+// IBMBlockCSIHostDefinitionSpec defines the observed state of HostDefinition
+type IBMBlockCSIHostDefinitionSpec struct {
```

Comment:
what is the difference between them? why do we need both?
I'm not sure `IBMBlockCSI` is relevant here

Reply:
it's allowing the that the structer will be for example `spec.HostDefinition.nodeName`

Reply:
I changed `IBMBlockCSIHostDefinitionSpec` name

Reply:
we already have [`HostDefinition `](https://github.com/IBM/ibm-block-csi-operator/blob/37cc4796fb43c8956c7b5f9a7b92aaa1e4e8b3cb/api/v1/hostdefinition_types.go#L61) type